src/components/Views/ObservationFields/ObservationFields.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+import React from 'react';
+import {
+  View,
+  Dimensions,
+  Keyboard,
+  KeyboardAvoidingView,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  TextInput,
+  Image,
+  ScrollView
+} from 'react-native';
+import { withNavigationFocus } from 'react-navigation';
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import CloseIcon from 'react-native-vector-icons/MaterialCommunityIcons';
+import FeatherIcon from 'react-native-vector-icons/Feather';
+import I18n from 'react-native-i18n';
+import FieldInput from '../../Base/FieldInput';
+import type { Field } from '../../../types/field';
+import type { Observation } from '../../../types/observation';
+import PencilIcon from '../../../images/editor-details.png';
+import {
+  LIGHT_GREY,
+  WHITE,
+  MANGO,
+  MEDIUM_GREY,
+  VERY_LIGHT_BLUE,
+  GREEN
+} from '../../../lib/styles';
+import Header from '../../Base/Header';
+
+export type StateProps = {
+  allFields: Field[],
+  selectedObservation: Observation
+};
+
+export type Props = {
+  isFocused: boolean,
+  navigation: any
+};
+
+export type DispatchProps = {
+  updateObservation: (o: Observation) => void,
+  goToPhotoView: (photoSource: string) => void,
+  addObservation: (o: Observation) => void,
+  goBack: () => void
+};
+
+type State = {};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: WHITE
+  },
+  progressContainer: {
+    height: 50,
+    backgroundColor: 'rgba(0, 0, 0, .8)',
+    flexDirection: 'row',
+    alignItems: 'center',
+    paddingHorizontal: 20
+  },
+  progressText: {
+    color: WHITE,
+    textAlign: 'center',
+    fontWeight: '400',
+    paddingRight: 25
+  }
+});
+
+I18n.fallbacks = true;
+I18n.translations = {
+  en: require('../../../translations/en'),
+  es: require('../../../translations/es')
+};
+
+class ObservationDetails extends React.Component<
+  Props & StateProps & DispatchProps,
+  State
+> {
+  constructor(props: StateProps & DispatchProps) {
+    super();
+    this.state = {};
+  }
+
+  shouldComponentUpdate(
+    nextProps: Props & StateProps & DispatchProps,
+    nextState: State
+  ) {
+    if (nextProps.isFocused) {
+      return nextProps !== this.props || nextState !== this.state;
+    }
+
+    return false;
+  }
+
+  handleSaveObservation = () => {
+    const { addObservation, selectedObservation } = this.props;
+
+    if (selectedObservation) {
+      addObservation({
```

Comment:
any reason why you spread `selectedObservation`?

Reply:
ooo you right, probs don't need to